examples/acoustics/acoustic.py

```diff
@@ -0,0 +1,179 @@
+
+import os
+import argparse
+
+import numpy as np
+import pandas as pd
+from matplotlib import pyplot as plt
+import matplotlib.ticker as ticker
+
+import flow360client
+
+def getCaseConfigParameters(caseId):
+    return flow360client.case.GetCaseInfo(caseId)['runtimeParams']
+
+def readAcousticCSVFile(acsv):
+    # read acoustic csv when the file is available via python api
+    # with tempfile.TemporaryDirectory() as tmpdir:
+    #     tmpfile = os.path.join(tmpdir, 'aeroacoustics_v3.csv')
+    #     flow360client.case.DownloadFile(caseId, 'aeroacoustics_v3.csv', tmpfile)
+    pressure_data = pd.read_csv(acsv,skipinitialspace=True)
+    times = pressure_data['physical_time']
+    steps = pressure_data['physical_step']
+    p = pressure_data.iloc[:,2:-1]
+    return [np.array(times),np.array(steps),np.array(p)]
+
+def filterZeroAcousticData(time,steps,pressure):
+    numObservers = pressure.shape[-1]
+    acoustic_data_indices = {}
+    acoustic_data = {}
+    # loop over observers to find where they are zero
+    for i in range(numObservers):
+        p_observer = pressure[:,i]
+        acoustic_data_indices[i] = np.where(p_observer!=0)[0]
+    first_elements = []
+    last_elements = []
+    # lists of elements with zeros at begin and end
+    for i in range(numObservers):
+        first_elements.append(acoustic_data_indices[i][0])
+        last_elements.append(acoustic_data_indices[i][-1])
+    # max index where still there is a zero element for all observers
+    start_index = max(first_elements)
+    # min index where still there is a zero element for all observers
+    end_index = min(last_elements)
+
+    # physical time and step where all observers all non zero
+    acoustic_data['physical_time'] = time[start_index:end_index]
+    acoustic_data['physical_step'] = steps[start_index:end_index]
+
+    # non-zero acoustic data for all observers with the same physical time
+    for i in range(numObservers):
+        p_observer = pressure[:,i]
+        acoustic_data[i] = p_observer[start_index:end_index]
+    return acoustic_data
```

Comment:
Also, i think we should have an option in this function to return the data per observer and process each observer indenpendently

Reply:
OK. Added an optional argument that lets the user define observerId as an input.

Reply:
When observerId is given as input it only removes zeros for that particular observer data and post-process that data alone.
When observerId is not given as input, it removes zeros based on the whole data and post-processed all observers.